src/components/table/columns.ts

```diff
@@ -49,18 +48,20 @@ export class ColumnDefinitionFactory {
 export const formatHeader = (column: Column) => (): string | HTMLElement => {
     const element = document.createElement('div');
     element.innerText = column.title;
-    if (!column.headerComponent) {
-        return;
-    }
 
-    const menu = document.createElement(column.headerComponent.name);
-    menu.setAttribute('class', 'tabulator-header-menu-button');
-
-    const props = column.headerComponent.props;
+    const header = document.createElement(column.headerComponent.name);
+    let props: object = column.headerComponent.props || {};
+    if (column.headerComponent.propsFactory) {
+        props = {
+            ...props,
+            // we pass null to propsFactory function because no data in column header
```

Comment:
We should clarify this in the interface documentation as well, that when the factory is used for header components there will be no data available